Sources/VaporToolbox/HerokuInit.swift

```diff
@@ -17,31 +17,44 @@ public final class HerokuInit: Command {
 
     public func run(arguments: [String]) throws {
 
-        func gitIsClean() throws {
-          do {
-              let status = try console.backgroundExecute(program: "git", arguments: ["status", "--porcelain"])
-              if status.trim() != "" {
-                  console.info("All current changes must be committed before running a Heroku init.")
-                  throw ToolboxError.general("Found uncommitted changes.")
-              }
-          } catch ConsoleError.backgroundExecute(_, _) {
-              throw ToolboxError.general("No .git repository found.")
-          }
+        func gitIsClean(log: Bool = true) throws {
+            do {
+                let status = try console.backgroundExecute(program: "git", arguments: ["status", "--porcelain"])
+                if status.trim() != "" {
+                    if log {
+                        console.info("All current changes must be committed before running a Heroku init.")
+                    }
+                    throw ToolboxError.general("Found uncommitted changes.")
+                }
+            } catch ConsoleError.backgroundExecute {
+                throw ToolboxError.general("No .git repository found.")
+            }
         }
 
         try gitIsClean()
 
         do {
+            let branches = try console.backgroundExecute(program: "git", arguments: ["branch"])
+            guard branches.contains("* master") else {
+                throw ToolboxError.general(
+                    "Please checkout master branch before initializing heroku. 'git checkout master'"
+                )
+            }
+        } catch ConsoleError.backgroundExecute(_, _, let message) {
```

Comment:
You probably want stderr here as well, though you'll need to verify you get the proper output you want. Some commands print errors to stderr or stdout or a combo.